src/java.base/linux/classes/sun/nio/ch/EPollSelectorImpl.java

```diff
@@ -125,7 +125,7 @@ protected int doSelect(Consumer<SelectionKey> action, long timeout)
                 if (numEntries == IOStatus.INTERRUPTED && timedPoll) {
                     // timed poll interrupted so need to adjust timeout
                     long adjust = System.nanoTime() - startTime;
-                    to -= TimeUnit.MILLISECONDS.convert(adjust, TimeUnit.NANOSECONDS);
+                    to =- (int) TimeUnit.NANOSECONDS.toMillis(adjust);
```

Comment:
This will now always assign a negative number to `to`.

--------------------------------------------------------------------------------

`=-` is not a compound assignment, it’s negation followed by a normal assignment.

Reply:
Well spotted, I don't think that change was intentionally.

Reply:
Ouch; Will fix:

I took Alan's earlier comment literally:

"This one can also be changed to:

to =- (int) TimeUnit.NANOSECONDS.toMillis(adjust);"